### PR TITLE
fix: add error handling for undefined dimensions and metrics in validation service

### PR DESCRIPTION
**Pull request by ZeRego**
Relates to: PROD-1990

### Description:
This PR adds error handling for undefined dimensions and metrics in the ValidationService. It captures these errors in Sentry to help diagnose issues where fields are unexpectedly undefined. 

The PR also improves the `getItemId` function by adding a check for undefined items and enhancing test coverage with new test cases that handle edge cases like items with undefined tables and names containing dots.

### CODE REVIEW

**Comment by ZeRego** [comment]
* **#18926** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/18926?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/lightdash/lightdash/18926?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by almeidabbm — APPROVED** [review]
Look good!!
FYI: this was related too, `changesets` could introduce undefined fields in explores - https://github.com/lightdash/lightdash/pull/18911

**Review by pimmey — APPROVED** [review]

File: packages/backend/src/services/ValidationService/ValidationService.ts [hunk 1]
```diff
@@ -667,6 +669,22 @@ export class ValidationService extends BaseService {
...
                 );
+
+                if (dimensions.find((d) => d === undefined)) {
+                    Sentry.captureException(
+                        new UnexpectedServerError(
+                            `Undefined dimension found in explore ${explore.name} in project ${projectUuid}`,
+                        ),
+                    );
+                }
+                if (metrics.find((m) => m === undefined)) {
```
Comment (almeidabbm) [inline packages/backend/src/services/ValidationService/ValidationService.ts:680]: you can also use `metrics.some`, but I checked and don't think it makes a big difference (if any) in terms of perf
